File: frontend/lib/api/pos.ts
import { apiClient } from './client'

export interface POSProduct {
  id: number
  name: string
  price: number
  stock: number
  sku?: string
  barcode?: string
  unit?: string
}

export interface CartItem {
  product_id: number
  product_name: string
  quantity: number
  unit_price: number
  subtotal: number
  stock_available: number
}

export interface CheckoutRequest {
  items: Array<{
    product_id: number
    quantity: number
    unit_price: number
  }>
  payment_method: 'cash' | 'mobile_money' | 'card' | 'credit'
  customer_name?: string
  customer_phone?: string
  discount?: number
  notes?: string
}

export interface CheckoutResponse {
  success: boolean
  sale_id: number
  invoice_id?: number
  invoice_number?: string
  total: number
  payment_method: string
}

export interface POSSession {
  id: number
  opened_at: string
  total_sales: number
  total_transactions: number
  cash_total: number
  mobile_money_total: number
  card_total: number
  credit_total: number
  is_active: boolean
}

export const posApi = {
  // Search products
  async searchProducts(query: string, limit: number = 20): Promise<POSProduct[]> {
    const response = await apiClient.get<POSProduct[]>('/pos/search', {
      params: { q: query, limit }
    })
    return response.data
  },

  // Validate cart
  async validateCart(items: Array<{ product_id: number; quantity: number; unit_price: number }>) {
    const response = await apiClient.post('/pos/cart/validate', { items })
    return response.data
  },

  // Checkout
  async checkout(data: CheckoutRequest): Promise<CheckoutResponse> {
    const response = await apiClient.post<CheckoutResponse>('/pos/checkout', data)
    return response.data
  },

  // Get POS session
  async getSession(): Promise<POSSession> {
    const response = await apiClient.get<POSSession>('/pos/session')
    return response.data
  },

  // Close session
  async closeSession() {
    const response = await apiClient.post('/pos/session/close')
    return response.data
  },
}

File: backend/app/schemas/pos.py
"""
POS schemas for request/response validation
"""
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class SaleItemCreate(BaseModel):
    product_id: int
    quantity: int
    unit_price: float


class CartItem(BaseModel):
    product_id: int
    product_name: str
    quantity: int
    unit_price: float
    subtotal: float
    stock_available: int


class CartUpdateRequest(BaseModel):
    items: List[SaleItemCreate]


class CheckoutRequest(BaseModel):
    items: List[SaleItemCreate]
    payment_method: str  # cash, mobile_money, card, credit
    customer_name: Optional[str] = None
    customer_phone: Optional[str] = None
    customer_id: Optional[int] = None  # Required if payment_method is "credit"
    discount: float = 0.0
    notes: Optional[str] = None


class SaleItemResponse(BaseModel):
    id: int
    product_id: int
    product_name: str
    quantity: int
    unit_price: float
    subtotal: float

    class Config:
        from_attributes = True


class SaleResponse(BaseModel):
    id: int
    total: float
    subtotal: float
    discount: float
    tax: float
    payment_method: str
    payment_status: str
    customer_name: Optional[str]
    customer_phone: Optional[str]
    created_at: datetime
    items: List[SaleItemResponse]

    class Config:
        from_attributes = True


class POSSessionResponse(BaseModel):
    id: int
    opened_at: datetime
    closed_at: Optional[datetime]
    total_sales: float
    total_transactions: int
    cash_total: float
    mobile_money_total: float
    card_total: float
    credit_total: float
    is_active: bool

    class Config:
        from_attributes = True

